code/game/objects/items/ego_weapons/zayin.dm

```diff
@@ -81,6 +274,30 @@
 	attack_verb_simple = list("slices", "slashes", "stabs")
 	hitsound = 'sound/weapons/bladeslice.ogg'
 
+	ability_cooldown_time = 30 SECONDS
+	matching_armor = /obj/item/clothing/suit/armor/ego_gear/letter_opener
+	use_message = "You use letter opener to send a message!"
+	use_sound = 'sound/items/handling/paper_drop.ogg'
+
+/obj/item/ego_weapon/support/letter_opener/Pulse(mob/user)
+	for(var/mob/living/carbon/human/M in GLOB.player_list)
+		var/msg = stripped_input(usr, "What do you wish to tell [M]?", null, "")
+		if(!msg)
+			return
+		/*log_directed_talk(user, M, msg, LOG_SAY, "[name]")*/
+		to_chat(M, "<span class='warning'>[user] has send you a message!</span>")
+		var/obj/item/paper/P = new(get_turf(M))
+		P.setText(msg)
+		P.icon_state = "mail"
+		QDEL_IN(P, 10 SECONDS)
+		to_chat(user, "<span class='boldnotice'>You transmit to [M]:</span> <span class='notice'>[msg]</span>")
+		for(var/ded in GLOB.dead_mob_list)
+			if(!isobserver(ded))
+				continue
+			var/follow_rev = FOLLOW_LINK(ded, user)
+			var/follow_whispee = FOLLOW_LINK(ded, M)
+			to_chat(ded, "[follow_rev] <span class='boldnotice'>[user] [name]:</span> <span class='notice'>\"[msg]\" to</span> [follow_whispee] <span class='name'>[M]</span>")
```

Comment:
Make this not itterate through EVERYONE individually.  Use something like this
```
var/choice = input(user,"Who would you like to message?","Select Someone") as null|anything in GLOB.player_list
if(!choice || !user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
    return
var/msg = stripped_input(usr, "What do you wish to tell [GLOB.player_list[choice]]?", null, "")
if(!msg)
    return
to_chat(M, "<span class='warning'>[user] has send you a message!</span>")
var/obj/item/paper/P = new(get_turf(M))
P.setText(msg)
P.icon_state = "mail"
QDEL_IN(P, 10 SECONDS)
to_chat(user, "<span class='boldnotice'>You transmit to [M]:</span> <span class='notice'>[msg]</span>")
for(var/ded in GLOB.dead_mob_list)
    if(!isobserver(ded))
        continue
    var/follow_rev = FOLLOW_LINK(ded, user)
    var/follow_whispee = FOLLOW_LINK(ded, M)
    to_chat(ded, "[follow_rev] <span class='boldnotice'>[user] [name]:</span> <span class='notice'>\"[msg]\" to</span> [follow_whispee] <span class='name'>[M]</span>")
```
To send a message to one person.